newrelic-agent/src/main/java/com/newrelic/agent/config/BaseConfig.java

```diff
@@ -239,6 +248,30 @@ public static List<String> getUniqueStringsFromString(String valuesString, Strin
         return result;
     }
 
+    /**
+     * Interprets a {@link String} as a {@link Map} using a format of 'key1:value1;key2:value2', with
+     * resulting String keys and values trimmed of whitespace.
+     * <p>
+     * Each parsed Map entry is passed to the consumer function, which may for example add the entry to
+     * a Map, or carry out validation.
+     * <p>
+     * If the String format cannot be parsed as a sequence of Map entries, or if the consumer throws a
+     * {@link ParseException} then a {@link ParseException} will be thrown.
+     */
+    public static void parseMapEntriesFromString(String mapAsString, MapParsingBiConsumer<String, String> parsedEntryHandler) throws
```

Comment:
This was in `LabelsConfigImpl` - I've shifted it up to this base class as a static method to align with the existing similar methods in this class eg `getUniqueStringsFromString`. 

This means that this logic is available to the flattened config access as well as the original typesafe style Configs (LabelsConfig etc)

However, I have taken out labels config specific validation, and just allowed it and other users, to pass in their own handler to do what they want with the parsed map entries.